blackbox_repository/blackbox_surrogate.py

```diff
@@ -79,14 +79,19 @@ def fit_surrogate(self, surrogate=KNeighborsRegressor(n_neighbors=1)) -> Blackbo
         # gets hyperparameters types, categorical for CategoricalHyperparameter, numeric for everything else
         numeric = []
         categorical = []
-        features = list(self.configuration_space.keys())
+
         if self.fidelity_space is not None:
-            features += self.fidelity_space.keys()
-        for hp in features:
+            surrogate_hps = dict()
+            surrogate_hps.update(self.configuration_space.items())
```

Comment:
no need for the items here, e.g. `surrogate_hps.update(self.configuration_space)` works.

Also I think those 3 lines are more straightforward to read:

```suggestion
            surrogate_hps = self.configuration_space
            if self.fidelity_space is not None:
              surrogate_hps.update(self.fidelity_space)
```